akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -576,7 +576,7 @@ object Uri {
     }
     def toMultiMap: Map[String, List[String]] = {
       @tailrec def append(map: Map[String, List[String]], q: Query): Map[String, List[String]] =
-        if (q.isEmpty) map else append(map.updated(q.key, q.value :: map.getOrElse(q.key, Nil)), q.tail)
+        if (q.isEmpty) map else append(map.updated(q.key, map.getOrElse(q.key, Nil) :+ q.value), q.tail)
```

Comment:
I know prepending is way more optimal than appending on `List`. I don't know if we need this performance or not.

From the Scaladoc:
> Performance
>
>Time: List has O(1) prepend and head/tail access. Most other operations are O(n) on the number of elements in the list. This includes the index-based lookup of elements, length, append and reverse.